modules/sops/default.nix

```diff
@@ -140,6 +145,15 @@ in {
         This option must be explicitly unset if <literal>config.sops.sshKeyPaths</literal>.
       '';
     };
+
+    activationMethod = mkOption {
+      type = types.enum [ "script" "systemd" ];
+      default = "script";
+      description = ''
+        Which method to use for setting up secrets. Use `script` for an
+        activation script, and `systemd` for a systemd unit.
```

Comment:
I think `systemd` is the way to go to use all features of sops-nix. I am thinking about deprecating the activation script.

Reply:
Should I change this PR to reflect this? I'd be happy to do it.

Reply:
I hope I get to hack next weekend again on sops-nix: https://github.com/systemd/systemd/pull/16568#issuecomment-720072151
I prepared systemd socket activation and with this PR https://github.com/systemd/systemd/pull/17510 I should be able to implement it properly and ready for networked KMS.

Reply:
To be more explicit with socket-based secret description it is easier to make systemd wait for secrets to become available without having to specify explicit dependencies. This was my major concern with having sops-nix running as a systemd service instead of activation script. I hope to be able to implement this soonish.